src/components/gcds-radio/gcds-radio.tsx

```diff
@@ -0,0 +1,127 @@
+
+@Component({
+  tag: 'gcds-radio',
+  styleUrl: 'gcds-radio.css',
+  shadow: false,
+  scoped: true,
+})
+export class GcdsRadio {
+
+  @Element() el: HTMLElement;
+
+  private lang: string;
+  private shadowElement?: HTMLInputElement;
+
+  /**
+   * Id attribute for an input element.
+   */
+  @Prop({ reflect: true, mutable: true }) radioId!: string;
+
+  /**
+   * Form field label
+   */
+  @Prop({ reflect: true, mutable: false }) label!: string;
+
+  /**
+   * Name attribute for an input element.
+   */
+  @Prop({ reflect: true, mutable: false }) name!: string;
+
+  /**
+   * Specifies if a form field is required or not.
+   */
+  @Prop({ reflect: true, mutable: false }) required: boolean;
+
+  /**
+   * Specifies if an input element is disabled or not.
+   */
+  @Prop({ reflect: true, mutable: true }) disabled: boolean;
+
+  /**
+   * Specifies if an input element is checked.
+   */
+  @Prop({ reflect: true, mutable: true }) checked: boolean;
+
+  /**
+   * Value for an input element.
+   */
+  @Prop({ reflect: true, mutable: false }) value: string;
+
+  /**
+   * Specifies if the input is invalid.
+   */
+  @Prop({ reflect: true, mutable: true }) hasError: boolean;
+  /**
+   * Hint displayed below the label.
+   */
+  @Prop({ reflect: true, mutable: false }) hint: string;
+
+  /**
+   * Emitted when the radio button is checked
+   */
+  @Event() gcdsRadioChange!: EventEmitter<void>;
+
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+  }
+
+  private onChange = (name) => {
+    this.gcdsRadioChange.emit(name);
+    this.checked = this.shadowElement.checked;
+  };
+
+  @Listen('gcdsRadioChange', { target: 'document' })
+  gcdsradioChangeEventHandler(event) {
+    if (event.detail == this.name && event.srcElement != this.shadowElement) {
+      if (this.checked) {
+        this.checked = false;
+      }
+    }
+  }
+
+  render() {
+    const { lang, radioId, label, name, required, disabled, value, checked, hint, hasError } = this;
+
+    const attrsInput = {
+      name,
+      disabled,
+      required,
+      value,
+      checked
+    };
+
+    if (hint) {
+      let hintID = hint ? `hint-${radioId}` : "";
+      attrsInput["aria-describedby"] = `${hintID}`;
+    }
+
+    if (hasError) {
+      attrsInput["aria-invalid"] = "true";
+    }
+
+    return (
+      <Host>
+        <div class={`gcds-radio-wrapper ${disabled ? 'gcds-disabled' : ''} ${hasError ? 'gcds-error' : ''}`}>
+          <input
+            id={radioId}
+            type="radio"
+            {...attrsInput}
+            onChange={() => this.onChange(name)}
+            ref={element => this.shadowElement = element as HTMLInputElement}
+          />
+          <gcds-label
+            label={label}
+            label-for={radioId}
+            lang={lang}
+          >
+          </gcds-label>
+          {hint ? <gcds-hint hint={hint} hint-id={radioId} />: null}
+        </div>
```

Comment:
Since the use case for radio buttons is always in a group, having the ability to assign an error message to a singular radio button isn't that useful. I am hoping to build a fieldset component that can help handle the validation and error handling of groups of inputs.

Reply:
True that, ignore this comment then :)